src/sessionTypes/machineAuth.js

```diff
@@ -101,8 +101,12 @@ class MachineAuth {
       return Promise.reject(new Error('No valid audience found'));
     }
 
-    if (!this._tokenPromise) {
-      this._tokenPromise = axios
+    if (!this.tokenPromises) {
+      this.tokenPromises = {};
```

Comment:
Maybe we should do this in the constructor?

Reply:
Yeah good thinking.